src/components/assignment/js/worlds-collection.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+var classFrom = require('../../../util/inherit');
+var Model = require('./world-model');
+var UserModel = require('./user-model');
+var serviceLocator = require('./service-locator');
+
+
+var Base = require('./base-collection');
+var __super = Base.prototype;
+
+var doneFn = function (dtd, after) {
+    return _.after(after, dtd.resolve);
+};
+
+var worldApi;
+
+module.exports = classFrom(Base, {
+    model: Model,
+
+    initialize: function () {
+        __super.initialize.apply(this, arguments);
+        worldApi = serviceLocator.worldApi();
+    },
+
+    autoAssignAll: function (options) {
+        return worldApi.autoAssign(options)
+            .then(function (worlds) {
+                this.reset(this.parse(worlds));
+            }.bind(this));
+    },
+
+    getIncompleteWorldsCount: function () {
+        return this.filter(function (w) {
+            return !w.get('complete');
+        }).length;
+    },
+
+    updateUser: function (user) {
+        var worldName = user.get('world');
+        var dtd = $.Deferred();
+        var prevWorld = this.getWorldByUser(user);
+        var curWorld = this.getOrCreateWorldById(worldName);
+        var done = doneFn(dtd, 1);
+
+        // check if there's anything to do
+        if (!prevWorld && !curWorld) {
+            return dtd.resolve().promise();
+        }
+
+        if (prevWorld) {
+            prevWorld.removeUser(user)
+                .then(function () {
+                    if (curWorld) {
+                        return curWorld.addUser(user);
+                    }
+                })
+                .then(done);
+        } else if (curWorld) {
+            curWorld.addUser(user)
+                .then(done);
+        }
+
+        return dtd.promise();
+    },
+
+    getOrCreateWorldById: function (worldName) {
```

Comment:
"createById" sounds weird. Maybe rename to "getByIdOrCreateWorld" or just "getOrCreateWorld"?